### PR TITLE
[ENHANCEMENT] Update `ember-cli-babel` to v8 in `app` and `addon` blueprints

### PR DESCRIPTION
**Pull request by bertdeblock**
Mainly opening to see how CI will respond.

Closes https://github.com/ember-cli/ember-cli/issues/9933.
Closes https://github.com/ember-cli/ember-cli/pull/9934.
~~Closes https://github.com/babel/ember-cli-babel/issues/453.~~

### CODE REVIEW

**Comment by bertdeblock** [comment]
https://github.com/embroider-build/embroider/pull/1343 should fix the failing Embroider runs.

File: blueprints/addon/index.js [hunk 1]
```diff
@@ -56,6 +56,11 @@ module.exports = {
     contents.dependencies['ember-cli-babel'] = contents.devDependencies['ember-cli-babel'];
     delete contents.devDependencies['ember-cli-babel'];
 
+    // Addons must bring in their own version of `@babel/core`
+    // when using `ember-cli-babel` >= v8.
+    contents.dependencies['@babel/core'] = contents.devDependencies['@babel/core'];
```
[the remark this answers is not in the document]
Comment (NullVoxPopuli) [reply blueprints/addon/index.js:61]: @bertdeblock provided some context here: https://github.com/babel/ember-cli-babel/blob/master/UPGRADING.md#upgrade-path-for-addons

tl;dr: v1 addons already don't care what the app is doing, as they compile themselves

**Review by NullVoxPopuli — APPROVED** [review]

**Comment by bertdeblock** [comment]
`ember-cli-babel` was accidentally bumped to v8 in the blueprints without adding the `@babel/core` dependency.
Closing this PR in favour of this bug fix => https://github.com/ember-cli/ember-cli/pull/10349.

File: blueprints/addon/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (bertdeblock) [reply blueprints/addon/index.js:61]: I've added that link as well over in https://github.com/ember-cli/ember-cli/pull/10349.
